app/controllers/application_controller.rb

```diff
@@ -60,7 +61,8 @@ def set_locale_from_subdomain
     if (I18n.locale != I18n.default_locale) && # Don’t redirect to en.crimethinc.com
        request.subdomain.empty? &&             # Don’t redirect if there’s a subdomain
        Rails.env.production?                   # Don’t redirect in development
-      return redirect_to subdomain: I18n.locale
+
+      return redirect_to({ subdomain: I18n.locale }.merge(params.permit!))
```

Comment:
ok, I think this works

Reply:
@veganstraightedge noticed you are looking at this PR.

I am not *sure* this works perfectly. need to do some more testing